ql/src/java/org/apache/hadoop/hive/ql/exec/tez/ReduceRecordProcessor.java

```diff
@@ -305,6 +305,19 @@ public void abort() {
     } else {
       LOG.info("reducer not setup yet. abort not being forwarded");
     }
+    if (mergeWorkList != null) {
+      for (BaseWork redWork : mergeWorkList) {
+        ((ReduceWork) redWork).getReducer().abort();
+      }
+    }
+    if (reduceWork != null) {
+      List<HashTableDummyOperator> dummyOps = reduceWork.getDummyOps();
+      if (dummyOps != null) {
+        for (Operator<?> dummyOp : dummyOps) {
+          dummyOp.abort();
+        }
+      }
+    }
```

Comment:
this implementation doesn't belong here, ReduceRecordProcessor is already overloaded, what about reduceWork.abort() and do the rest there